site/src/lib/config.ts

```diff
@@ -5,4 +5,6 @@ export const FRONTEND_DOMAIN =
 
 export const USE_HTTPS = process.env.NODE_ENV !== "development";
 
-export const FRONTEND_URL = `http${USE_HTTPS ? "" : "s"}//${FRONTEND_DOMAIN}`;
+export const FRONTEND_URL = `http${USE_HTTPS ? "s" : ""}://${FRONTEND_DOMAIN}`;
```

Comment:
I think `USE_HTTPS` is unnecessary - we could just have a `FRONTEND_URL` or `FRONTEND_ORIGIN` which includes the protocol, and turn on secure cookies if it starts with `https`.

The current implementation would also prevent (a) running a production build locally without HTTPS enabled, (b) running a development build locally with HTTPS enabled.

Reply:
My only issue with this would is that it would make it harder to determine the `FRONTEND_DOMAIN`, as we now have to parse `FRONTEND_URL` for this. But I suppose the following would work:
```TypeScript
const FRONTEND_DOMAIN = (new URL(FRONTEND_URL)).hostname;
```

Reply:
Are you suggesting we do the following to prevent breaking deployments?
```TypeScript
export const FRONTEND_URL = process.env.NEXT_PUBLIC_VERCEL_URL
  ? `https://${process.env.NEXT_PUBLIC_VERCEL_URL}`
  : process.env.FRONTEND_URL ?? "localhost:3000";
```

Reply:
Yes, assuming  `VERCEL_URL` doesn't include the protocol

Reply:
It doesn't 👍